### ISSUE TITLE
Better support for offline / low connectivity during first time flow

### ISSUE THREAD

**Issue by danjm** [MetaMask/metamask-extension#5210]
During the first time flow, if users can't connect to a network, they get an infinite spinner that blocks them from further action.

We should allow users to do as much as possible without any network connection, while continuously attempting to connect in the background.

Once connection is absolutely necessary, and if Metamask is still attempting to connect, we should give the user an action to take (even if that is just 'please wait')

**Comment by bdresser** [comment]
relates to 
https://github.com/MetaMask/metamask-extension/issues/5143
https://github.com/MetaMask/metamask-extension/issues/4496
https://github.com/MetaMask/metamask-extension/issues/2636
https://github.com/MetaMask/metamask-extension/issues/1483

as a bare minimum we shouldn't show the "Connecting..." spinner on top of onboarding or the unlock screen

**Comment by bdresser** [comment]
noting this has come up a couple times through support and I've experienced it myself several times this week on the unlock screen.

**Comment by cjeria** [comment]
Add a persistent orange banner that says "You are offline" when user is not online.

**Comment by bdresser** [comment]
relates to https://github.com/MetaMask/metamask-extension/pull/5854 as well

**Comment by bschorchit** [comment]
This is currently fixed for default networks, but not for custom networks so I'm keeping this open.

**Comment by holantonela** [comment]
> Add a persistent orange banner that says "You are offline" when user is not online.

I like this idea, and it also give us room to contemplate the spectrum of connectivity users can experience (which is not only boolean).

I'm exploring options here. They all include users' ability to continue using the app with a non dismissable warning.

https://www.figma.com/file/qmtRKfT3K9DBFZEYFwsS6f/Offline-screen?node-id=102%3A38&t=GHvRl7Q3VUAGyKoi-1

**Comment by holantonela** [comment]
Mobile related issue https://github.com/MetaMask/metamask-mobile/issues/3433

**Comment by masonreid1729** [comment]
Yes, there’s some kind of givеаway going on. I got some Ethereum there yesterday for different wallets. It’s unlikely they’ll take it back from me, but it’s a pity I didn’t have more wallets and couldn’t get more. But maybe it’s for the best.

![Image](https://github.com/user-attachments/assets/5d17aaa9-3e59-45be-9cde-4b60c9704510)
